### DISCUSSION TITLE
Initial state of "errors" with "isValid"

### DISCUSSION THREAD

**Discussion by vikharytonov** [react-hook-form/react-hook-form#10172] [Q&A]
Hi all, could some one assist with understanding of logic for initial state of ```errors``` vs ```isValid```.
When we have initialize form with ```useForm``` we get ```errors``` and ```isValid``` from ```formState```.
```
const REQUIRED_MSG = "This field is required";
const validationSchema = object({
  name: string().min(7).required(REQUIRED_MSG),
  lastName: string().min(10).required(REQUIRED_MSG),
});
const {
  formState: { errors, isValid },
  register,
  trigger,
  getFieldState,
} = useForm({
  defaultValues: { name: "john", lastName: "doe" },
  resolver: yupResolver(validationSchema),
});

  ```
I'm wondering why we have simple flag of valid or not valid state of the form and not able to get the list of errors for this form as initial values? And to get the list of errors we need to trigger validation.
You can take a look at code [here](https://codesandbox.io/s/react-hook-form-initial-state-of-error-bz2rqd) to check initial state with different ```defaultValues```

**Comment by leapful** [comment]
@vikharytonov 

The reason is simple, it's the default behavior of form validation for humanity, for example, login form asking username and password will never show required error on first load when end-user did not touch anything yet, or, like a Update Profile page will never validate its loaded initial data from BE as the data must be always valid before being stored in the BE database.

react-hook-form applies this same behavior because validation error should be shown when end-user interact with the form, as it is used to notify end-user that there is something wrong with their input. If the initial values (a.k.a default values) are not correct at the beginning then it's the responsibility of developer to correct it, not end-user.
